src/main/java/org/jabref/gui/entryeditor/SourceTab.java

```diff
@@ -243,4 +250,17 @@ private void storeSource(BibEntry outOfFocusEntry, String text) {
             LOGGER.debug("Incorrect source", ex);
         }
     }
+
+    @Override
+    public void highlightPattern(Optional<Pattern> highlightPattern) {
+        if (highlightPattern.isPresent()) {
+            Matcher matcher = highlightPattern.get().matcher(codeArea.getText());
+            while (matcher.find()) {
+                for (int i = 0; i <= matcher.groupCount(); i++) {
+                    //TODO: Styling doesn't work
+                    codeArea.setStyle(matcher.start(), matcher.end(), Collections.singletonList("-fx-text-fill: red;"));
```

Comment:
@tobiasdiez  Do you have any idea why the styling doesn't work? Text should be highlighted red.
I also tried the example from here with a style class, but did not work either;
https://github.com/FXMisc/RichTextFX#codearea

Reply:
Is the code executed correctly, ie does the matcher finds the query text?

Reply:
Yes, I debugged it and the matcher correctly finds the text. The method for styling is called correctly. Matcher start and end are also correct. 
I also tried adding the code to the css and use the set Style Class but same result.

Reply:
If this one is not working then I have also no idea https://github.com/FXMisc/RichTextFX/wiki/Text-Styles%3A-Inline%2C-Style-Class%2C-and-Custom-Style-Objects#style-class-pattern

Reply:
Actually for code Area it's a bit different. Got it to work now:
https://github.com/FXMisc/RichTextFX/wiki/RichTextFX-CSS-Reference-Guide#issue-with-styleclassed-based-css-default-styling